### PR TITLE
Update DG v1.1

### PR DESCRIPTION
**Pull request by yuechen2001**
Update Developer Guide 

- [x] Target user profile
- [x] Value proposition
- [x] User stories
- [x] Use cases

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -274,67 +337,148 @@ _{Explain here how the data archiving feature will be implemented}_
...
+| `*`      | New user                                | have a tutorial feature that shows me how to use the app                                                          |                                                                         |
+| `* *`    | Student attending conferences or events | quickly exchange contact information with fellow attendees through QR codes                                       | quickly add new contacts                                                |
+| `* *`    | Team Lead                               | have access to integrated online learning platforms to track the courses or certificates completed by my contacts | understand their evolving skills                                        |
+| `* * *`  | Student                                 | endorse or rate the skills of my contacts                                                                         | keep track of their expertise and choose the right team members         |
```
Comment (angsongyee) [inline docs/DeveloperGuide.md:381]: Ideally should be ordered by priority but not a big deal

Comment (angsongyee) [inline docs/DeveloperGuide.md:381]: Ideally should be ordered by priority level but not a big deal

**Review by angsongyee — APPROVED** [review]
Okay with merging

**Review by maiyasaliha — APPROVED** [review]
LGTM

**Pull request merged**